pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -884,6 +865,52 @@ public SuccessResponse deleteSegments(
     }
   }
 
+  @DELETE
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/segments/{tableName}/select")
```

Comment:
@wirybeaver We deprecated `GET /segments/{tableName}/select` over `GET /segments`. (We added the annotation). 

Can we add these filters to `DELETE /segments` API?

Reply:
@snleee I actually copy the parameters from `GET /segments/{tableName}`, and enforce `GET /segments/{tableName}` and this delete function use the same private method selectSegments. It's just a coincidence that the path overlap with the deprecated `GET /segments/{tableName}/select`. I can rename the path to `DELETE /segments/{tableName}/choose`.

The `DELETE /segments/{tableName}` has annotation `@ApiOperation(value = "Delete all segments", notes = "Delete all segments")`, that's the reason why I want to have a different path.

Reply:
@wirybeaver 
We can change the convention to be:

`Delete all segments -> delete all segments that satisfies the filter`.

`startTime, endTime, excludeReplacedSegments`. Please take a look at `/segments/{tableName}`. It already has those filters for fetching segments.

We can keep the backward compatibility by picking all segments if no filter is provided. How do you think?

Reply:
From the perspective of safety, since it's a delete api, my hunt is to enforce users MUST input non-empty startTime and endTime, which will break the compatibility that delete all segment if no filter is provides. That's another reason to have different PATH. User would know they are doing a dangerous operation if they use `DELETE /segments/{tableName}` meanwhile `DELETE /segments/{tableName}/choose` has a safety guard even though they forgot to add the timestamp.

I am OK to add filters into `DELETE /segments/{tableName}` and remove `DELETE /segments/{tableName}/choose` if you thought the lack of prevention of missing time window is not a major concern.

Reply:
@snleee Just a gentle reminder for your thoughts.

Reply:
@snleee @jtao15 Could you share your idea when you are available?

Reply:
@wirybeaver sorry for the late response on this. 

The current behavior for `DELETE /segments/{tableName}` API is already to delete all segments in the table.

Adding filter (start, end) means that we will delete the segments matching the filter. I think that we are keeping the backward compatibility and not break any behavior?

Am I missing something?

Reply:
I wanted to enforce both start and end to be non empty to avoid the case that customers incidentally delete all segments due to forgetting inputing (start, end). However, since start and end must be non empty, current `DELETE /segments/{tableName}` will fail, which is a breaking change. If you thought we don't need to validate empty start and end, I am good to reuse the existing api

Reply:
I see. You need the protection. In that case, I'm good with keeping the separate API. It would be nice to have if 'DELETE /segments' also gets extra filter just to have the feature parity with `GET /segments`. Anyway, this can be addressed as part of the separate PR.

Reply:
Will have a cosmetic change.use `DELETE /segments/{tableName}/choose` instead of `DELETE /segments/{tableName}/select`